analyzers/src/SonarAnalyzer.CSharp/Rules/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -0,0 +1,116 @@
+            invocation.Expression is MemberAccessExpressionSyntax memberAccess
+            && memberAccess.GetName() is { } name
+            && NamesWithTypes.ContainsKey(name)
+            && IsInvocationOn(memberAccess, semanticModel, NamesWithTypes[name]);
+
+        private static bool IsInvocationOn(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel, ImmutableArray<KnownType> knownTypes) =>
+            semanticModel.GetSymbolInfo(memberAccess).Symbol is { } symbol
+            && symbol.ContainingType.DerivesFromAny(knownTypes);
+
+        private static bool HasEmptyPasswordArgument(SeparatedSyntaxList<ArgumentSyntax> argumentList) =>
+            ConnectionStringArgument(argumentList)?.Expression switch
+            {
+                LiteralExpressionSyntax literal when literal.IsKind(SyntaxKind.StringLiteralExpression) => IsVulnerable(literal.Token.ValueText) && !HasSanitizers(literal.Token.ValueText),
+                InterpolatedStringExpressionSyntax interpolatedString => HasEmptyPasswordAndNoSanitizers(interpolatedString),
+                BinaryExpressionSyntax binaryExpression when binaryExpression.IsKind(SyntaxKind.AddExpression) => HasEmptyPasswordAndNoSanitizers(binaryExpression),
+                _ => false
+            };
+
+        // First search a named argument, then search literals, then fallback on the first argument (for constant propagation check).
+        // This is an easy way to support explicit extension method invocation.
+        private static ArgumentSyntax ConnectionStringArgument(SeparatedSyntaxList<ArgumentSyntax> argumentList) =>
+            // Where(cond).First() is more efficient than First(cond)
+            argumentList.Where(a => a.NameColon?.Name.Identifier.ValueText == "connectionString").FirstOrDefault()
+                ?? argumentList.Where(a => a.Expression.IsAnyKind(SyntaxKind.StringLiteralExpression, SyntaxKind.InterpolatedStringExpression, SyntaxKind.AddExpression)).FirstOrDefault()
+                ?? argumentList.FirstOrDefault();
+
+        // For both interpolated strings and concatenation chain, it's easier to search in the string representation of the tree, rather than doing string searches for each individual
+        // string token inside.
+        private static bool HasEmptyPasswordAndNoSanitizers(ExpressionSyntax expression)
+        {
+            var toString = expression.ToString();
+            return IsVulnerable(toString) && !HasSanitizers(toString);
+        }
+
+        private static bool IsVulnerable(string connectionString) =>
+            connectionString.EndsWith("Password=")
+            || connectionString.Contains("Password=;")
+            // this is an edge case, for a string interpolation or concatenation the toString() will contain the ending "
+            // we prefer to keep it like this for the simplicity of the implementation
+            || connectionString.EndsWith("Password=\"");
```

Comment:
this wasn't necessary in the initial implementation, but I prefered to simplify the implementation (see commit 2c59138) and keep this "hack" as a toll